### PR TITLE
TransmittingPatch 

### PR DESCRIPTION
**Pull request by Undid-Iridium**
https://github.com/Exiled-Team/Exiled-EA/pull/202

### CODE REVIEW

File: Exiled.Events/Patches/Events/Player/Transmitting.cs [hunk 1]
```diff
@@ -56,20 +55,24 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
 
             ListPool<CodeInstruction>.Shared.Return(newInstructions);
         }
 
-        private static bool HandleTransmitting(ReferenceHub hub, VoiceModuleBase voiceModule)
+        private static bool HandleTransmitting(PersonalRadioPlayback radioPlayback)
         {
-            if (hub == null || Player.Get(hub) is not Player player)
+            ReferenceHub hub = radioPlayback._owner;
+            if (hub == null || Player.Get(hub) is not Player player || Server.Host.ReferenceHub == hub)
```
Comment (IRacle1) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:69]: `Player::Get` return null for host hub, third check it useless

**Review by IRacle1 — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Player/Transmitting.cs [hunk 2]
```diff
@@ -45,9 +47,6 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                     // hub
                     new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsFrom(newInstructions[index]),
 
-                    // voiceModule
-                    new(OpCodes.Ldloc_1),
-
                     // HandleTransmitting(ReferenceHub, VoiceModule)
                     new(OpCodes.Call, Method(typeof(Transmitting), nameof(HandleTransmitting))),
```
Comment (IRacle1) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:51]:
```suggestion
                    // HandleTransmitting(PersonalRadioPlayback)
                    new(OpCodes.Call, Method(typeof(Transmitting), nameof(HandleTransmitting))),
```

File: Exiled.Events/Patches/Events/Player/Transmitting.cs [hunk 3]
```diff
@@ -45,9 +47,6 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                     // hub
                     new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsFrom(newInstructions[index]),
```
Comment (IRacle1) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:48]:
```suggestion
                    // this
                    new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsFrom(newInstructions[index]),
```

File: Exiled.Events/Patches/Events/Player/Transmitting.cs [hunk 4]
```diff
@@ -56,20 +55,24 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                 });
 
             // -2 to return false
-            newInstructions[newInstructions.Count - 2].WithLabels(retLabel);
+            newInstructions[newInstructions.Count - 1].WithLabels(retLabel);
```
Comment (IRacle1) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:58]: -2?

**Review by IRacle1 — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Player/Transmitting.cs (hunk 3 above)
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Player/Transmitting.cs:48]: Ignore the comments in my drafts.., I will fix, my concern is logic :( - I have <2 hours of sleep again 🙏

File: Exiled.Events/Patches/Events/Player/Transmitting.cs (hunk 1 above)
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Player/Transmitting.cs:69]: bet

File: Exiled.Events/Patches/Events/Player/Transmitting.cs (hunk 4 above)
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Player/Transmitting.cs:58]: Copy paste.. and that was used in a different function so -2 made sense back then

Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Player/Transmitting.cs:58]: image
 
 Interesting, github can't see change

File: Exiled.Events/Patches/Events/Player/Transmitting.cs (hunk 3 above)
Comment (IRacle1) [reply Exiled.Events/Patches/Events/Player/Transmitting.cs:48]: 😭

File: Exiled.Events/Patches/Events/Player/Transmitting.cs (hunk 4 above)
Comment (IRacle1) [reply Exiled.Events/Patches/Events/Player/Transmitting.cs:58]: git moment

Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Player/Transmitting.cs:58]: 😢

File: Exiled.Events/Patches/Events/Player/Transmitting.cs [hunk 5]
```diff
@@ -60,20 +54,24 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
 
-        private static bool HandleTransmitting(ReferenceHub hub, VoiceModuleBase voiceModule)
+        private static bool HandleTransmitting(PersonalRadioPlayback radioPlayback)
         {
-            if (hub == null || Player.Get(hub) is not Player player)
+            ReferenceHub hub = radioPlayback._owner;
+            if (hub == null || Player.Get(hub) is not Player player || Server.Host.ReferenceHub == hub)
+            {
                 return false;
+            }
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:71]:
```suggestion
            ReferenceHub hub = radioPlayback._owner;

            if (hub == null || Player.Get(hub) is not Player player || Server.Host.ReferenceHub == hub)
                return false;
```

File: Exiled.Events/Patches/Events/Player/Transmitting.cs [hunk 6]
```diff
@@ -60,20 +54,24 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
 
-            TransmittingEventArgs ev = new(player, voiceModule);
+            IVoiceRole voiceModule = player.RoleManager.CurrentRole as IVoiceRole;
+            TransmittingEventArgs ev = new(player, voiceModule?.VoiceModule);
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:74]:
```suggestion
            IVoiceRole voiceRole = player.RoleManager.CurrentRole as IVoiceRole;
            TransmittingEventArgs ev = new(player, voiceRole == null ? null : voiceRole.VoiceModule);
```

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Player/Transmitting.cs [hunk 7]
```diff
@@ -29,7 +26,7 @@ namespace Exiled.Events.Patches.Events.Player
     ///     Patches <see cref="PersonalRadioPlayback.IsTransmitting(ReferenceHub)" />.
     ///     Adds the <see cref="Handlers.Player.Transmitting" /> event.
     /// </summary>
```
Comment (IRacle1) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:28]:
```suggestion
    ///     Patches <see cref="PersonalRadioPlayback.Update()" />.
    ///     Adds the <see cref="Handlers.Player.Transmitting" /> event.
    /// </summary>
```

**Review by IRacle1 — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Player/Transmitting.cs (hunk 6 above)
Comment (IRacle1) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:74]: you can remove `as` operator, and use IVoiceRole cast directly
```suggestion
            IVoiceRole voiceModule = (IVoiceRole)player.RoleManager.CurrentRole;
            TransmittingEventArgs ev = new(player, voiceModule.VoiceModule);
```

**Review by IRacle1 — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Player/Transmitting.cs (hunk 6 above)
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Player/Transmitting.cs]: Rider disagree's so I am going with IRacles suggestion

**Review by Thundermaker300 — APPROVED** [review]

File: Exiled.Events/Patches/Events/Player/Transmitting.cs [hunk 8]
```diff
@@ -60,20 +54,23 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                 });
 
             // -2 to return false
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:56]:
```suggestion
```

File: Exiled.Events/Patches/Events/Player/Transmitting.cs [hunk 9]
```diff
@@ -60,20 +54,23 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
         {
-            if (hub == null || Player.Get(hub) is not Player player)
+            ReferenceHub hub = radioPlayback._owner;
+
+            if (hub == null || Player.Get(hub) is not Player player || Server.Host.ReferenceHub == hub)
                 return false;
 
-            TransmittingEventArgs ev = new(player, voiceModule);
+            IVoiceRole voiceModule = (IVoiceRole)player.RoleManager.CurrentRole;
+            TransmittingEventArgs ev = new(player, voiceModule.VoiceModule);
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/Transmitting.cs:73]:
```suggestion
            TransmittingEventArgs ev = new(player, ((IVoiceRole)player.RoleManager.CurrentRole).VoiceModule);
```

**Review by iopietro — CHANGES_REQUESTED** [review]

**Review by iopietro — APPROVED** [review]

**Comment by iopietro** [comment]
It seems that two old conversations marked as resolved haven't been resolved yet.

**Review by iopietro — APPROVED** [review]

**Review by Thundermaker300 — APPROVED** [review]

**Pull request merged**